app/cdash/include/common.php

```diff
@@ -914,19 +914,54 @@ function create_aggregate_build($build, $siteid = null): Build
 }
 
 /**
- * Extract a tarball to a given directory.
+ * Extract a tarball within the local storage directory.
  */
-function extract_tar(string $filename, string $dirName): bool
+function extract_tar(string $stored_filepath): string
 {
+    if (!Storage::exists($stored_filepath)) {
+        Log::error("{$stored_filepath} does not exist", [
+            'function' => 'extract_tar',
+        ]);
+        return '';
```

Comment:
What do you think about throwing an exception instead?  That's probably the best way to handle error cases here, but I'd be fine deferring that to a future PR if you think it's too far outside the scope of this PR.